samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManagerUtil.java

```diff
@@ -411,4 +438,292 @@ public static Set<String> getSideInputStoreNames(
     }
     return sideInputStores;
   }
+
+  /**
+   * Inits and Restores all the task stores.
+   * Note: In case of {@link BlobStoreRestoreManager}, this method retries init and restore with getDeleted flag if it
+   * receives a {@link DeletedException}. This will create a new checkpoint for the corresponding task.
+   */
+  public static CompletableFuture<Map<TaskName, Checkpoint>> initAndRestoreTaskInstances(
```

Comment:
Minor: Suggest moving all of these to a new Util class, e.g.  ContainerStorageManagerRestoreUtil